src/main/java/org/opentripplanner/routing/algorithm/raptor/router/RaptorRouter.java

```diff
@@ -76,16 +81,16 @@ public TripPlan route(RoutingRequest request) {
 
         double startTimeRouting = System.currentTimeMillis();
 
-        // TODO Time zones
-        int departureTime = Instant.ofEpochMilli(request.dateTime * 1000).atZone(ZoneId.systemDefault()).toLocalTime().toSecondOfDay();
+        int departureTime = secondsSinceStartOfTime(otpRRDataProvider.getStartOfTime(), request.getDateTime().toInstant());
 
         // TODO Expose parameters
         // TODO Remove parameters from API
         RequestBuilder builder = new RequestBuilder();
         builder.profile(RangeRaptorProfile.MULTI_CRITERIA)
                 .searchParams()
                 .earliestDepartureTime(departureTime)
-                .searchWindowInSeconds(40  * 60)
+                .searchWindowInSeconds(request.raptorSearchWindow)
+                .latestArrivalTime(departureTime + 12 * 60 * 60)
```

Comment:
`latestArrivalTime` needs to have a value so that the reverse search heuristic has valid parameters. I'm unsure of what a valid default would be.

Reply:
This is "work in progress", the ARRIVE_BY search is not implemented jet, and I am unsure how well it works. The relationship between `earliestDepartureTime`(EAT), `searchWindow`(SW) and `latestArrivalTime`(LAT) is that:
- EAT and SW should give you a depart after search. The LAT is optional (needs more work).
- LAT and SW should give you an arrive by search - not implemented, there is a few things that need to be done to compleate this.
- EAT and LAT should give you a "Timetable" search. 

I think this needs to be a bit more clear from the API design - And maybe we should enforce some of it. 

@flaktack Thank you for addressing this, there is a list of work to be done here: https://github.com/entur/r5/issues (This will be migrated to r5 or OTP at some point).

Reply:
@t2gran thanks for the explanation :+1: